e2e-tests/playwright/tests/functional/channels/drafts/drafts_on_deleted_message.spec.ts

```diff
@@ -0,0 +1,181 @@
+    await lastPostByAdminInChannel.openRHSWithPostOptions();
+
+    // # Write a message as a user
+    const sidebarRight = channelPage.sidebarRight;
+    await sidebarRight.toBeVisible();
+    await sidebarRight.postMessage('Replying to a thread');
+    await sidebarRight.sendMessage();
+
+    // # Write a message in the reply thread but don't send it now so that it becomes a draft
+    const draftMessageByUser = 'I should be in drafts by User';
+    await sidebarRight.postMessage(draftMessageByUser);
+
+    // # Bring focus back to center channel textbox, so that the draft is saved
+    await channelPage.postMessage('');
+
+    // * Verify drafts link in channel sidebar is visible
+    await channelPage.sidebarLeft.draftsExist();
+
+    // # Delete the last post by admin
+    try {
+        await adminClient.deletePost(adminPost.id);
+    } catch (error) {
+        throw new Error('Failed to delete post by admin');
+    }
+
+    // * Verify drafts in user's textbox is still visible
+    const rhsTextboxValue = await sidebarRight.getTextboxValue();
+    expect(rhsTextboxValue).toBe(draftMessageByUser);
+
+    // # Click on remove post
+    const deletedPostByAdminInRHS = await sidebarRight.getRHSPostById(adminPost.id);
+    await deletedPostByAdminInRHS.clickOnRemovePost();
+
+    // * Verify the drafts links should also be removed from sidebar
+    await channelPage.sidebarLeft.draftsDoesntExist();
+});
+
+test('MM-T5435_2 Gloabl Drafts link in sidebar should be hidden when user deletes root post ', async ({pw, pages}) => {
+    const {team, user, userClient} = await pw.initSetup();
+
+    try {
+        // # Set preferences to disable tooltips for drafts and threads
+        await userClient.savePreferences(user.id, createNoTooltipForDraftsAndThreadsPreferences(user.id));
+    } catch (error) {
+        throw new Error('Failed to set preferences');
+    }
+
+    // # Log in as user in new browser context
+    const {page} = await pw.testBrowser.login(user);
+
+    // # Visit default channel page
+    const channelPage = new pages.ChannelsPage(page);
+    await channelPage.goto(team.name);
+    await channelPage.toBeVisible();
+
+    // # Post a message in the channel
+    await channelPage.postMessage('Message which will be deleted');
+    await channelPage.sendMessage();
+
+    // # Start a thread by clicking on reply button
+    const lastPost = await channelPage.getLastPost();
+    await lastPost.openRHSWithPostOptions();
+
+    const sidebarRight = channelPage.sidebarRight;
+    await sidebarRight.toBeVisible();
+
+    // # Write a message in the thread
+    await sidebarRight.postMessage('Replying to a thread');
+    await sidebarRight.sendMessage();
+
+    // # Write a message in the reply thread but don't send it
+    await sidebarRight.postMessage('I should be in drafts');
+
+    // # Bring focus back to center channel textbox, so that the draft in RHS is saved
+    await channelPage.postMessage('');
```

Comment:
This failed in `ipad`. One option is to close the RHS by close button. You may add the locator at `ChannelsSidebarRight`, then this could be:
```javascript
await sidebarRight.closeButton.click();
```

Reply:
Done ✅